### PR TITLE
Integrate boot file make into top level

### PR DESCRIPTION
**Pull request by mmlb**
## Description

Lift the rules from installer/alpine/Makefile into rules.mk.j2 so that the top-level make can handle building the alpine files itself. Also cleanup/fix the Dockerfile & build.sh files that are responsible for building the Alpine boot files.

## Why is this needed

Extracted out of #72 for better reviewability.

## How Has This Been Tested?
Builds kernel, initramfs, and modloop files as needed.

### CODE REVIEW

File: installer/alpine/Dockerfile [hunk 1]
```diff
@@ -1,100 +1,101 @@
...
+ENV KERNEL 4.14.55
+ENV MAKEFLAGS -j$(($(nproc) * 2))
+# make sure PKGREL is +1 what is in APKGBUILD
+ENV PKGREL 1
+
+RUN true && \
+    # Setup self-signed keys to sign the built packages \
+    abuild-keygen -a -i -n && \
+    # Fetch alpine package tree \
+    curl -fL https://github.com/alpinelinux/aports/archive/$COMMIT.tar.gz | tar -zxf - && \
```
Comment (DavidHwu) [inline installer/alpine/Dockerfile:55]: do we care about failure scenarios ?

Comment (mmlb) [reply installer/alpine/Dockerfile:55]: More than handling via pipefail? https://github.com/tinkerbell/osie/pull/79/files#diff-b67911656ef5d18c4ae36cb6741b7965R2

File: rules.mk.j2 [hunk 2]
```diff
@@ -220,23 +220,27 @@ else
 	ci/vm.sh network_test | tee build/$@.log >/dev/$T
 endif
 
+ifneq ($(CI),drone)
+build/osie-aarch64.tar.gz: /proc/sys/fs/binfmt_misc/qemu-aarch64
+endif
 build/osie-aarch64.tar.gz: SED=/FROM/ s|.*|FROM multiarch/ubuntu-debootstrap:arm64-xenial|
-build/osie-x86_64.tar.gz: SED=
+build/osie-x86_64.tar.gz:  SED=
```
Comment (splaspood) [inline rules.mk.j2:227]: extra space, no?

File: rules.mk.j2 [hunk 3]
```diff
@@ -220,23 +220,27 @@ else
...
-	$(Q)docker save osie:$* > $@.tmp
-	$(Q)mv $@.tmp $@
+	$(Q)docker save osie:$* > $@
 
+ifneq ($(CI),drone)
+build/osie-runner-aarch64.tar.gz: /proc/sys/fs/binfmt_misc/qemu-aarch64
+endif
 build/osie-runner-aarch64.tar.gz: SED=/FROM/ s|.*|FROM multiarch/alpine:arm64-v3.7|
-build/osie-runner-x86_64.tar.gz: SED=
+build/osie-runner-x86_64.tar.gz:  SED=
```
Comment (splaspood) [inline rules.mk.j2:239]: another extra space

**Review by splaspood — CHANGES_REQUESTED** [review]

File: installer/alpine/Dockerfile (hunk 1 above)
Comment (DavidHwu) [reply installer/alpine/Dockerfile:55]: My question was geared around what error handling strategy we use... as I see this in docker file I understand now.  Withdrawing.

File: rules.mk.j2 (hunk 2 above)
Comment (mmlb) [reply rules.mk.j2:227]: lol, @mikemrm said the same thing. I did it on purpose to line up the SEDs in the Makefile, it was annoying me to be unaligned.

File: rules.mk.j2 (hunk 3 above)
Comment (mmlb) [reply rules.mk.j2:239]: ditto

**Review by DavidHwu — DISMISSED** [review]

**Review by splaspood — DISMISSED** [review]

**Comment by mmlb** [comment]
@splaspood PTAL at the last 2 commits. These setup a way to skip building the Alpine files and adds some updated docs to the README.

**Review by splaspood — DISMISSED** [review]

**Comment by mmlb** [comment]
Trying to figure out why drone/make is still trying to rebuild the alpine assets.

**Comment by mmlb** [comment]
Finally got the "skip building alpine assets" stuff correct.

@splaspood @DavidHwu one more looksie please.

File: apps/runner.sh [hunk 4]
```diff
@@ -86,9 +86,14 @@ phone_home_url=$(jq -r .phone_home_url "$metadata")
 trap fail EXIT
 
 service docker start
-until docker info; do
+i=0
+# shellcheck disable=SC2169
+until docker info >/dev/null || [[ $i -gt 5 ]]; do
```
Comment (splaspood) [inline apps/runner.sh:91]: I feel like this 'wait for docker to be ready' code that's duplicated should be pulled out into a shared function...  but not strongly enough to block the review on.

File: apps/workflow-helper.sh [hunk 5]
```diff
@@ -42,11 +44,14 @@ mkdir -p /etc/docker/ /etc/docker/certs.d/ "/etc/docker/certs.d/$docker_registry
 cp ca.pem "/etc/docker/certs.d/$docker_registry/ca.crt"
 
 service docker start
-
-until docker info; do
-	echo 'Waiting for docker to respond...'
+i=0
+# shellcheck disable=SC2169
+until docker info >/dev/null || [[ $i -gt 5 ]]; do
```
Comment (splaspood) [inline apps/workflow-helper.sh:49]: see above

File: apps/osie-installer.sh [hunk 6]
```diff
@@ -135,9 +133,15 @@ cat <<-EOF | grep -v '^\s*$'
 	${packet_base_url:+packet_base_url: $packet_base_url}
 EOF
 
-until docker info; do
+service docker start
+i=0
+# shellcheck disable=SC2169
+until docker info >/dev/null || [[ $i -gt 5 ]]; do
```
Comment (splaspood) [inline apps/osie-installer.sh:139]: see below

File: installer/alpine/Dockerfile [hunk 7]
```diff
@@ -1,100 +1,98 @@
...
-
-RUN sed -i 's/silentoldconfig/olddefconfig/g' APKBUILD && \
-    echo 'CONFIG_KEXEC=y' >> config-vanilla.x86_64 && \
+ENV COMMIT bb52b0d3f60cd75197ef218362f193378dfd869d
+ENV FLAVOR vanilla
+ENV KERNEL 4.14.55
+# make sure PKGREL is +1 what is in APKGBUILD
+ENV PKGREL 1
+
+RUN true && \
```
Comment (splaspood) [inline installer/alpine/Dockerfile:51]: true && ?

File: shell.nix [hunk 8]
```diff
@@ -30,14 +30,15 @@ let
...
+    cpio
     curl
     docker
     git
     git-lfs
     gnumake
     gnused
     libarchive
-    minio
+    minio-client
```
Comment (splaspood) [inline shell.nix:41]: Is this even used anymore?

**Review by splaspood — CHANGES_REQUESTED** [review]

File: apps/osie-installer.sh (hunk 6 above)
Comment (splaspood) [reply apps/osie-installer.sh:139]: doh, I figured it put these comments in order of the code, not in the order that I left them...  anyway.. you get the idea ;)

File: apps/runner.sh (hunk 4 above)
Comment (mmlb) [reply apps/runner.sh:91]: Would agree but this other file would also have to be fetched from /init (really we should just embed them into the initramfs) which I kind of don't really want to do.

File: apps/workflow-helper.sh (hunk 5 above)
Comment (mmlb) [reply apps/workflow-helper.sh:49]: ditto

File: apps/osie-installer.sh (hunk 6 above)
Comment (mmlb) [reply apps/osie-installer.sh:139]: heh

File: installer/alpine/Dockerfile (hunk 7 above)
Comment (mmlb) [reply installer/alpine/Dockerfile:51]: I want to keep the comment above the command but docker won't let me start a `RUN` with a comment so I just used `true` for that.

File: shell.nix (hunk 8 above)
Comment (mmlb) [reply shell.nix:41]: actually yes, we use it to upload to s3. We could use a different tool, but there isn't really a need to change it in this PR.

Comment (splaspood) [reply shell.nix:41]: ahh, ok.

**Review by splaspood — APPROVED** [review]

**Pull request merged**